int64 fromRobotID
int64 toRobotID